common/djangoapps/third_party_auth/strategy.py

```diff
@@ -34,3 +34,29 @@ def setting(self, name, default=None, backend=None):
         # At this point, we know 'name' is not set in a [OAuth2|LTI|SAML]ProviderConfig row.
         # It's probably a global Django setting like 'FIELDS_STORED_IN_SESSION':
         return super(ConfigurationModelStrategy, self).setting(name, default, backend)
+
+    def request_host(self):
+        """
+        Host in use for this request
+        """
+        # TODO: this override is a temporary measure until upstream python-social-auth patch is merged:
+        # https://github.com/omab/python-social-auth/pull/741
+        if self.setting('RESPECT_X_FORWARDED_HEADERS', False):
```

Comment:
@e0d I believe there are at least two approaches that more or less comply with this comment:

* `if not self.setting('IGNORE_X_FORWARDED_HEADERS', False):`
* `if self.setting('RESPECT_X_FORWARDED_HEADERS', True):`

Which one suits best? I would take the second approach, as it reduces the code changes that needs to be done (and backported to solutions fork that already uses this setting in production).

To save some time, I'll implement the second approach now.